### PR TITLE
feat(query): support of Terraform modules for "Auto Scaling Group Wit…

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
…h No Associated ELB"

**Check why this is failing in unit tests**

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

**Comment by rogeriopeixotocx** [comment]
@tiagocruz20 this query was reverted and reopened as PR due to failing checks.

Please add a new `OR` check in the policy, verifying if the `aws_autoscaling_group` has load_balancers > 0: 
```
count(resource.load_balancers) > 0
```

File: assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego [hunk 1]
```diff
@@ -29,5 +29,38 @@ CxPolicy[result] {
...
+
+	not common_lib.valid_key(module, keyToCheck)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("module[%s].load_balancers", [name]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": "'load_balancers' is set and not empty",
+		"keyActualValue": "'load_balancers' is undefined",
+		"searchLine": common_lib.build_search_line(["module", name, "load_balancers"], []),
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego:48]:
```suggestion
		"searchLine": common_lib.build_search_line(["module", name], []),
```

File: assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego [hunk 2]
```diff
@@ -29,5 +29,38 @@ CxPolicy[result] {
...
+
+CxPolicy[result] {
+	module := input.document[i].module[name]
+	keyToCheck := common_lib.get_module_equivalent_key("aws", module.source, "aws_autoscaling_group", "load_balancers")
+
+	not common_lib.valid_key(module, keyToCheck)
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("module[%s].load_balancers", [name]),
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego:44]:
```suggestion
		"searchKey": sprintf("module[%s]", [name]),
```

File: assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego [hunk 3]
```diff
@@ -29,5 +29,38 @@ CxPolicy[result] {
...
+CxPolicy[result] {
+	module := input.document[i].module[name]
+	keyToCheck := common_lib.get_module_equivalent_key("aws", module.source, "aws_autoscaling_group", "load_balancers")
+
+	count(module[keyToCheck]) == 0
+
+	result := {
+		"documentId": input.document[i].id,
+		"searchKey": sprintf("module[%s].load_balancers", [name]),
+		"issueType": "MissingAttribute",
```
Comment (felipe-avelar) [inline assets/queries/terraform/aws/auto_scaling_group_with_no_associated_elb/query.rego:61]:
```suggestion
		"issueType": "IncorrectValue",
```

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check suggestions. Also, first rule is missing `searchLine`

**Review by joaoReigota1 — APPROVED** [review]
LGTM

**Pull request merged**